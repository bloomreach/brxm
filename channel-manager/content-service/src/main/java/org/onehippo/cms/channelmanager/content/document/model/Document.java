/*
 * Copyright 2016-2019 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * This bean represents a document, stored in the CMS. It can be serialized into JSON to expose it through a REST API.
 */
@JsonInclude(Include.NON_NULL)
public class Document {
    private String id;                // UUID
    private String variantId;
    private String branchId;
    private String displayName;
    private String urlName;
    private String repositoryPath;
    private DocumentInfo info;        // read-only information about (the current state of) the document
    private Map<String, List<FieldValue>> fields;
    private String state;

    public Document() {
        setInfo(new DocumentInfo());
        setFields(new HashMap<>());
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getVariantId() {
        return variantId;
    }

    public void setVariantId(final String variantId) {
        this.variantId = variantId;
    }

    public String getBranchId() {
        return branchId;
    }

    public void setBranchId(final String branchId) {
        this.branchId = branchId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(final String displayName) {
        this.displayName = displayName;
    }

    public String getUrlName() {
        return urlName;
    }

    public void setUrlName(final String urlName) {
        this.urlName = urlName;
    }

    public String getRepositoryPath() {
        return repositoryPath;
    }

    public void setRepositoryPath(final String repositoryPath) {
        this.repositoryPath = repositoryPath;
    }

    public DocumentInfo getInfo() {
        return info;
    }

    public void setInfo(final DocumentInfo info) {
        this.info = info;
    }

    public Map<String, List<FieldValue>> getFields() {
        return fields;
    }

    public void setFields(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public String getState() {
        return state;
    }

    public void setState(final String state) {
        this.state = state;
    }
}
