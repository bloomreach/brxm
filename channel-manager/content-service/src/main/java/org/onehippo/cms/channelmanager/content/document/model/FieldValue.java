/*
 * Copyright 2016-2018 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;

import org.onehippo.cms.channelmanager.content.documenttype.field.validation.ValidationErrorInfo;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * FieldValue encapsulates the value of a field.
 *
 * If the field is primitive, its value is stored in the "value" field. If it is a compound, the child
 * values are stored in the "fields" map. If a field value has validation errors, they can be associated
 * with the field value by storing them inside the errorInfo. The ID field is for future use: when the
 * client would like to reorder fields, each field value must have an ID to recognize a changed value order.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class FieldValue {
    private String value;                         // Stringified value for primitive fields
    private String id;                            // UUID of the field, if needed
    private Map<String, Object> metadata;         // Optional metadata
    private Map<String, List<FieldValue>> fields; // Subfields of compound
    private String chosenId;                      // ID of a choice in case of a choice field
    private FieldValue chosenValue;               // Singular value of a choice field

    // private String id;    // ID for reordering, unique within a level (list) of field values
    private ValidationErrorInfo errorInfo;

    public FieldValue() { }

    public FieldValue(final String value) {
        this.value = value;
    }

    public FieldValue(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public FieldValue(final String choiceId, final FieldValue choiceValue) {
        this.chosenId = choiceId;
        this.chosenValue = choiceValue;
    }

    public Optional<String> findValue() {
        return Optional.ofNullable(value);
    }

    public boolean hasValue() {
        return value != null;
    }

    // use for serialization and testing only, to avoid NPE.
    public String getValue() {
        return value;
    }

    public void setValue(final String value) {
        this.value = value;
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public Map<String, Object> getMetadata() {
        return metadata;
    }

    public void setMetadata(final Map<String, Object> metadata) {
        this.metadata = metadata;
    }

    public Optional<Map<String, List<FieldValue>>> findFields() {
        return Optional.ofNullable(fields);
    }

    // use for serialization and testing only, to avoid NPE.
    public Map<String, List<FieldValue>> getFields() {
        return fields;
    }

    public void setFields(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public Optional<String> findChosenId() {
        return Optional.ofNullable(chosenId);
    }

    // use for serialization and testing only, to avoid NPE.
    public String getChosenId() {
        return chosenId;
    }

    public void setChosenId(final String chosenId) {
        this.chosenId = chosenId;
    }

    public Optional<FieldValue> findChosenValue() {
        return Optional.ofNullable(chosenValue);
    }

    // use for serialization and testing only, to avoid NPE.
    public FieldValue getChosenValue() {
        return chosenValue;
    }

    public void setChosenValue(final FieldValue chosenValue) {
        this.chosenValue = chosenValue;
    }

    public ValidationErrorInfo getErrorInfo() {
        return errorInfo;
    }

    public void setErrorInfo(final ValidationErrorInfo errorInfo) {
        this.errorInfo = errorInfo;
    }

    public boolean hasErrorInfo() {
        return errorInfo != null;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        final FieldValue that = (FieldValue) o;

        // only include the 'value' fields so the comparison ignores different meta-data of different variants
        return Objects.equals(value, that.value) &&
                Objects.equals(fields, that.fields) &&
                Objects.equals(chosenId, that.chosenId) &&
                Objects.equals(chosenValue, that.chosenValue);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value, fields, chosenId, chosenValue);
    }
}
