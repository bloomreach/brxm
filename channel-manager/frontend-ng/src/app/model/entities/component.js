/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as HstConstants from '../constants';
import { ComponentEntity } from './component-entity';

export class Component extends ComponentEntity {
  hasExperiment() {
    return !!this.getExperimentId();
  }

  getExperimentId() {
    return this._meta[HstConstants.EXPERIMENT_ID];
  }

  getExperimentStateLabel() {
    if (!this.hasExperiment()) {
      return null;
    }

    return `EXPERIMENT_LABEL_${this._meta[HstConstants.EXPERIMENT_STATE]}`;
  }

  getContainer() {
    return this.container;
  }

  setContainer(container) {
    this.container = container;
  }

  getType() {
    return 'component';
  }

  getRenderVariant() {
    return this._meta[HstConstants.RENDER_VARIANT] || HstConstants.DEFAULT_RENDER_VARIANT;
  }

  getReferenceNamespace() {
    return this._meta[HstConstants.REFERENCE_NAMESPACE];
  }
}
