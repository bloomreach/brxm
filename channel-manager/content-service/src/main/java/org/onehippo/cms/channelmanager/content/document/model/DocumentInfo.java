/*
 * Copyright 2016-2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.List;
import java.util.Objects;

import org.onehippo.cms.channelmanager.content.documenttype.model.DocumentType;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * This bean carries information of a document, stored in the CMS.
 * It is part of a document and can be serialized into JSON to expose it through a REST API.
 * Type {@code type} attribute refers to the document's {@link DocumentType} by id.
 */
@JsonInclude(JsonInclude.Include.NON_DEFAULT)
// only return the error count and error messages, never let clients set it
@JsonIgnoreProperties(value = {"errorCount", "errorMessages"}, allowGetters = true)
public class DocumentInfo {

    /**
     * enveloped reference to document type: { id: "namespace:typename" }
     */
    private Type type;

    // whether this document has auto-drafted changes that have not been saved to the preview variant yet
    private boolean dirty;

    // the number of validation errors
    private int errorCount;

    // human-readable messages of document-level validation errors
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private List<String> errorMessages;

    // maps to hippostd:publishableSummary (new, live, changed, or unknown)
    private PublicationState publicationState;

    private boolean canPublish;
    private boolean canRequestPublication;
    private String locale;
    private boolean canKeepDraft;
    private boolean retainable;

    public boolean isRetainable() {
        return retainable;
    }

    public Type getType() {
        return type;
    }

    public void setTypeId(final String id) {
        type = new Type(id);
    }

    public void setCanKeepDraft(final boolean saveDraft) {
        this.canKeepDraft = saveDraft;
    }

    public void setRetainable(final boolean retainable) {
        this.retainable = retainable;
    }

    public static class Type {

        private final String id;

        @JsonCreator
        public Type(@JsonProperty("id") String id) {
            this.id = id;
        }

        public String getId() {
            return id;
        }
    }

    public boolean isDirty() {
        return dirty;
    }

    public void setDirty(final boolean dirty) {
        this.dirty = dirty;
    }

    public int getErrorCount() {
        return errorCount;
    }

    public void setErrorCount(final int errorCount) {
        this.errorCount = errorCount;
    }

    public List<String> getErrorMessages() {
        return errorMessages;
    }

    public void setErrorMessages(final List<String> errorMessages) {
        this.errorMessages = errorMessages;
    }

    public PublicationState getPublicationState() {
        return publicationState;
    }

    public void setPublicationState(final PublicationState publicationState) {
        this.publicationState = publicationState;
    }

    public boolean isCanPublish() {
        return canPublish;
    }

    public boolean isCanKeepDraft(){
        return canKeepDraft;
    }

    public boolean isCanRequestPublication() {
        return canRequestPublication;
    }

    public void setCanPublish(final boolean canPublish) {
        this.canPublish = canPublish;
    }

    public void setCanRequestPublication(final boolean canRequestPublication) {
        this.canRequestPublication = canRequestPublication;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(final String locale) {
        this.locale = locale;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof DocumentInfo)) {
            return false;
        }
        final DocumentInfo that = (DocumentInfo) o;
        return isDirty() == that.isDirty() &&
                getErrorCount() == that.getErrorCount() &&
                isCanPublish() == that.isCanPublish() &&
                isCanRequestPublication() == that.isCanRequestPublication() &&
                isCanKeepDraft() == that.isCanKeepDraft() &&
                isRetainable() == that.isRetainable() &&
                Objects.equals(getType(), that.getType()) &&
                Objects.equals(getErrorMessages(), that.getErrorMessages()) &&
                getPublicationState() == that.getPublicationState() &&
                Objects.equals(getLocale(), that.getLocale());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getType(), isDirty(), getErrorCount(), getErrorMessages(), getPublicationState(), isCanPublish(), isCanRequestPublication(), getLocale(), isCanKeepDraft(), isRetainable());
    }


    @Override
    public String toString() {
        return "DocumentInfo{" +
                "type=" + type +
                ", dirty=" + dirty +
                ", errorCount=" + errorCount +
                ", errorMessages=" + errorMessages +
                ", publicationState=" + publicationState +
                ", canPublish=" + canPublish +
                ", canRequestPublication=" + canRequestPublication +
                ", locale='" + locale + '\'' +
                ", canKeepDraft=" + canKeepDraft +
                ", retainable=" + retainable +
                '}';
    }
}
