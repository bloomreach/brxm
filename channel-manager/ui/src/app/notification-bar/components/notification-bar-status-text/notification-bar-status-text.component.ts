/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { DatePipe } from '@angular/common';
import { Component, Input } from '@angular/core';

import { DocumentState } from '../../../models/document-state.enum';
import { XPageState } from '../../../models/xpage-state.model';
import { XPageStatus } from '../../../models/xpage-status.enum';
import { PageService } from '../../../services/page.service';

@Component({
  selector: 'em-notification-bar-status-text',
  templateUrl: 'notification-bar-status-text.component.html',
})
export class NotificationBarStatusTextComponent {
  @Input()
  set state(value: XPageState) {
    this.text = this.getText(value);
    this.textParams = {
      pageName: value.name,
      status: this.getDocumentStatusText(value.state),
      dateTime: this.datePipe.transform(value.scheduledRequest?.scheduledDate, 'full'),
    };
  }

  text: string | undefined;
  textParams: { pageName: string, status: string | undefined, dateTime: string | null } | undefined;

  constructor(
    private readonly pageService: PageService,
    private readonly datePipe: DatePipe,
  ) {}

  private getText(state: XPageState): string | undefined {
    const status = this.pageService.getXPageStatus(state);

    if (!status) {
      this.text = undefined;
      return;
    }

    const statusTextMap = {
      [XPageStatus.Published]: 'NOTIFICATION_BAR_XPAGE_LABEL_LIVE',
      [XPageStatus.Offline]: 'NOTIFICATION_BAR_XPAGE_LABEL_OFFLINE',
      [XPageStatus.UnpublishedChanges]: 'NOTIFICATION_BAR_XPAGE_LABEL_LIVE_UNPUBLISHED_CHANGES',
      [XPageStatus.PublicationRequest]: 'NOTIFICATION_BAR_XPAGE_LABEL_PUBLICATION_REQUESTED',
      [XPageStatus.RejectedRequest]: 'NOTIFICATION_BAR_XPAGE_LABEL_REQUEST_REJECTED',
      [XPageStatus.TakeOfflineRequest]: 'NOTIFICATION_BAR_XPAGE_LABEL_TAKE_OFFLINE_REQUESTED',
      [XPageStatus.ScheduledPublication]: 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION',
      [XPageStatus.ScheduledToTakeOffline]: 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE',
      [XPageStatus.ScheduledPublicationRequest]: 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION_REQUESTED',
      [XPageStatus.ScheduledToTakeOfflineRequest]: 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE_REQUESTED',
    };

    return statusTextMap[status];
  }

  private getDocumentStatusText(state: DocumentState): string | undefined {
    switch (state) {
      case DocumentState.Live: return 'life';
      case DocumentState.Unpublished: return 'unpublished changes';
      case DocumentState.New: return 'offline';
    }
  }
}
