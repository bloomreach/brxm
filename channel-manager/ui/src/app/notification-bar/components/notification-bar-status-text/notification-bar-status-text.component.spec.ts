/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NO_ERRORS_SCHEMA, Pipe, PipeTransform } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TranslateService } from '@ngx-translate/core';

import { DocumentState } from '../../../models/document-state.enum';
import { XPageStatusInfo } from '../../../models/page-status-info.model';
import { XPageStatus } from '../../../models/xpage-status.enum';
import { MomentPipe } from '../../../shared/pipes/moment.pipe';

import { NotificationBarStatusTextComponent } from './notification-bar-status-text.component';

@Pipe({
  name: 'translate',
})
class TranslateMockPipe implements PipeTransform {
  transform(value: string, params: { pageName: string, status: string | undefined, dateTime: string | null } | undefined): string {
    return `${value}${params?.pageName}${params?.status}${params?.dateTime}`;
  }
}

describe('NotificationBarStatusTextComponent', () => {
  let component: NotificationBarStatusTextComponent;
  let fixture: ComponentFixture<NotificationBarStatusTextComponent>;

  beforeEach(() => {
    const momentPipe = {
      transform: jest.fn(v => v),
    };

    const translateServiceMock = {
      instant: jest.fn(v => v),
    };

    fixture = TestBed.configureTestingModule({
      declarations: [
        NotificationBarStatusTextComponent,
        TranslateMockPipe,
      ],
      providers: [
        { provide: MomentPipe, useValue: momentPipe },
        { provide: TranslateService, useValue: translateServiceMock },
      ],
      schemas: [NO_ERRORS_SCHEMA],
    }).createComponent(NotificationBarStatusTextComponent);

    component = fixture.componentInstance;
  });

  describe.each([
    ['Published', new XPageStatusInfo(
      XPageStatus.Published,
      DocumentState.Live,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_LIVE'],
    ['Offline', new XPageStatusInfo(
      XPageStatus.Offline,
      DocumentState.New,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_OFFLINE'],
    ['UnpublishedChanges', new XPageStatusInfo(
      XPageStatus.UnpublishedChanges,
      DocumentState.Changed,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_LIVE_UNPUBLISHED_CHANGES'],
    ['PublicationRequest', new XPageStatusInfo(
      XPageStatus.PublicationRequest,
      DocumentState.Unpublished,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PUBLICATION_REQUESTED'],
    ['RejectedRequest', new XPageStatusInfo(
      XPageStatus.RejectedRequest,
      DocumentState.Unpublished,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_REQUEST_REJECTED'],
    ['TakeOfflineRequest', new XPageStatusInfo(
      XPageStatus.TakeOfflineRequest,
      DocumentState.Live,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_TAKE_OFFLINE_REQUESTED'],
    ['ScheduledPublication', new XPageStatusInfo(
      XPageStatus.ScheduledPublication,
      DocumentState.Unpublished,
      'some page',
      1596811323,
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION'],
    ['ScheduledToTakeOffline', new XPageStatusInfo(
      XPageStatus.ScheduledToTakeOffline,
      DocumentState.Live,
      'some page',
      1596811323,
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE'],
    ['ScheduledPublicationRequest', new XPageStatusInfo(
      XPageStatus.ScheduledPublicationRequest,
      DocumentState.Unpublished,
      'some page',
      1596811323,
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION_REQUESTED'],
    ['ScheduledToTakeOfflineRequest', new XPageStatusInfo(
      XPageStatus.ScheduledToTakeOfflineRequest,
      DocumentState.Live,
      'some page',
      1596811323,
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE_REQUESTED'],
    ['ProjectInProgress', new XPageStatusInfo(
      XPageStatus.ProjectInProgress,
      DocumentState.Live,
      'some page',
      undefined,
      'project name',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_PROGRESS'],
    ['ProjectInReview', new XPageStatusInfo(
      XPageStatus.ProjectInReview,
      DocumentState.Live,
      'some page',
      undefined,
      'project name',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_REVIEW'],
    ['ProjectPageApproved', new XPageStatusInfo(
      XPageStatus.ProjectPageApproved,
      DocumentState.Live,
      'some page',
      undefined,
      'project name',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_APPROVED'],
    ['ProjectPageRejected', new XPageStatusInfo(
      XPageStatus.ProjectPageRejected,
      DocumentState.Live,
      'some page',
      undefined,
      'project name',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_REJECTED'],
    ['ProjectRunning', new XPageStatusInfo(
      XPageStatus.ProjectRunning,
      DocumentState.Live,
      'some page',
      undefined,
      'project name',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IS_RUNNING'],
    ['EditingSharedContainers', new XPageStatusInfo(
      XPageStatus.EditingSharedContainers,
      DocumentState.Live,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_EDITING_SHARED_CONTAINERS'],
    ['PreviousVersion', new XPageStatusInfo(
      XPageStatus.PreviousVersion,
      DocumentState.Live,
      'some page',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_PREVIOUS_VERSION'],
    ['Locked', new XPageStatusInfo(
      XPageStatus.Locked,
      DocumentState.Live,
      'some page',
      undefined,
      undefined,
      undefined,
      'username',
    ), 'NOTIFICATION_BAR_XPAGE_LABEL_LOCKED_BY_USER'],
  ])('if xpage status is %s', (statusName, statusInfo, expectedText) => {
    beforeEach(() => {
      component.statusInfo = statusInfo;

      fixture.detectChanges();
    });

    test(`should provide ${expectedText} label`, () => {
      expect(component.text).toBe(expectedText);
    });

    test(`should render ${expectedText} label`, () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
