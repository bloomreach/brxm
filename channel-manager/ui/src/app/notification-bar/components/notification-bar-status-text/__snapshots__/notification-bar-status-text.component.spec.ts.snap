// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBarStatusTextComponent if xpage status is EditingSharedContainers should render NOTIFICATION_BAR_XPAGE_LABEL_EDITING_SHARED_CONTAINERS label 1`] = `
<div
  id="root33"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_EDITING_SHARED_CONTAINERSsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is Locked should render NOTIFICATION_BAR_XPAGE_LABEL_LOCKED_BY_USER label 1`] = `
<div
  id="root37"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_LOCKED_BY_USERsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is NotPartOfProject should render NOTIFICATION_BAR_XPAGE_LABEL_NOT_PART_OF_PROJECT label 1`] = `
<div
  id="root21"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_NOT_PART_OF_PROJECTsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is Offline should render NOTIFICATION_BAR_XPAGE_LABEL_OFFLINE label 1`] = `
<div
  id="root3"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_OFFLINEsome pageOFFLINEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is PreviousVersion should render NOTIFICATION_BAR_XPAGE_LABEL_PREVIOUS_VERSION label 1`] = `
<div
  id="root35"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PREVIOUS_VERSIONsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ProjectInProgress should render NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_PROGRESS label 1`] = `
<div
  id="root23"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_PROGRESSsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ProjectInReview should render NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_REVIEW label 1`] = `
<div
  id="root25"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IN_REVIEWsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ProjectPageApproved should render NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_APPROVED label 1`] = `
<div
  id="root27"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_APPROVEDsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ProjectPageRejected should render NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_REJECTED label 1`] = `
<div
  id="root29"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_PAGE_REJECTEDsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ProjectRunning should render NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IS_RUNNING label 1`] = `
<div
  id="root31"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PROJECT_IS_RUNNINGsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is PublicationRequest should render NOTIFICATION_BAR_XPAGE_LABEL_PUBLICATION_REQUESTED label 1`] = `
<div
  id="root7"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_PUBLICATION_REQUESTEDsome pageUNPUBLISHED_CHANGESundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is Published should render NOTIFICATION_BAR_XPAGE_LABEL_LIVE label 1`] = `
<div
  id="root1"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_LIVEsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is RejectedRequest should render NOTIFICATION_BAR_XPAGE_LABEL_REQUEST_REJECTED label 1`] = `
<div
  id="root9"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_REQUEST_REJECTEDsome pageUNPUBLISHED_CHANGESundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ScheduledPublication should render NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION label 1`] = `
<div
  id="root13"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATIONsome pageUNPUBLISHED_CHANGES1596811323

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ScheduledPublicationRequest should render NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION_REQUESTED label 1`] = `
<div
  id="root17"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_PUBLICATION_REQUESTEDsome pageUNPUBLISHED_CHANGES1596811323

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ScheduledToTakeOffline should render NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE label 1`] = `
<div
  id="root15"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINEsome pageLIVE1596811323

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is ScheduledToTakeOfflineRequest should render NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE_REQUESTED label 1`] = `
<div
  id="root19"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_SCHEDULED_TO_TAKE_OFFLINE_REQUESTEDsome pageLIVE1596811323

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is TakeOfflineRequest should render NOTIFICATION_BAR_XPAGE_LABEL_TAKE_OFFLINE_REQUESTED label 1`] = `
<div
  id="root11"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_TAKE_OFFLINE_REQUESTEDsome pageLIVEundefined

</div>
`;

exports[`NotificationBarStatusTextComponent if xpage status is UnpublishedChanges should render NOTIFICATION_BAR_XPAGE_LABEL_LIVE_UNPUBLISHED_CHANGES label 1`] = `
<div
  id="root5"
>
  
  
   NOTIFICATION_BAR_XPAGE_LABEL_LIVE_UNPUBLISHED_CHANGESsome pageCHANGEDundefined

</div>
`;
